regicide_game/player.py

```diff
@@ -0,0 +1,65 @@
+import random
+from typing import List
+
+from regicide_game.card import Card
+
+
+class Player:
+    hand: List(Card)
+
+    def __init__(self, max_hand_size, tavern_deck) -> None:
+        """
+        Deal cards to a player
+        """
+        if len(tavern_deck) > 0:
+            for i in range(min(max_hand_size, len(tavern_deck))):
+                self.hand.append(tavern_deck.pop(0))
+
+    def play_card(self) -> Card:
+        """
+        Plays a random card from the hand.
+        """
+        if len(self.hand) > 0:
+            played_card = random.choice(self.hand)
+            self.hand.remove(played_card)
+
+        return played_card
+
+    def discard_card(self, card_to_discard, discard_pile):
+        """
+        Discards a selected card to the discard pile.
+        """
+        self.hand.remove(card_to_discard)
+        discard_pile.append(card_to_discard)
+
+    def suffer_damage(self, enemy_card, discard_pile) -> bool:
+        """
+        Discard cards from the hand with a total value at least equal to the enemy's attack value.
+
+        Strategy of discarding: from left to right (simple, refine in the future)
+
+        Returns true if the player successfully takes damage, false if the game is over!
+        """
+        # determine the damage that will be dealt to you by the enemy
+        damage = enemy_card.power
+
+        # check if the damage is higher than the sum of all your cards' health
+        player_health = 0
+        for card in self.hand:
+            player_health += card.health
```

Comment:
I think we decided that `health` was only the enemy health.
If that is the case, then this should actually be `power`.

I thought it was:
- `power` the damage this enemy deals to players, the damage it deals to enemies and the amount that it protects you for (for Jack this is 10)
- `health` the damage you need to deal to this enemy card to defeat it (for Jack this is 20)